staging/src/k8s.io/kms/pkg/util/util.go

```diff
@@ -52,3 +58,105 @@ func ParseEndpoint(endpoint string) (string, error) {
 
 	return u.Path, nil
 }
+
+func PollImmediateUntilWithContext(ctx context.Context, interval time.Duration, condition func(context.Context) (done bool, err error)) error {
```

Comment:
Added a comment above. The previous review comments suggested using a simple inline for loop with sleep instead of copying code from apimachinery/util.

Reply:
add comments inline.